rfcs/007 - Alternative Ways to Create Mappers.md

```diff
@@ -0,0 +1,122 @@
+# 007 - Alternative Ways to Create Mappers
+
+## Current Status
+
+### Proposed
+
+2023-02-07
+
+### Accepted
+
+YYYY-MM-DD
+
+#### Approvers
+
+- Full Name <email@example.com>
+
+### Implementation
+
+- [ ] Implemented: [One or more PRs](https://github.com/iasql/iasql-engine/some-pr-link-here) YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author(s)
+
+- Mohammad Teimori Pabandi <m@iasql.com>
+
+## Summary
+
+Having Pure-SQL modules would help us expand more. One key advantage is that we don't need a PR on our engine to add coverage for a new service, we just use a syntax like this and everything JustWorks™.
+```postgresql
+SELECT iasql_install('https://github.com/someModuleDeveloper/awsEc2Module');
+```
+That's where we started, but it might be useful to assess different paths we might take for creation of an IaSQL module/mapper.
+
+## Proposal
+
+To create a module, we need a mapper. I'll continue this RFC with my proposed solution for mappers, since a module is a set of mappers, and entities (with a bit of oversimplification).
+
+First, let's see what we need to create a mapper:
+- A cloud id function
+- An equality checker function
+- An entity, that the mapper manages it
```

Comment:
Specifically for the line `An entity, that the mapper manages it` this is a good thing to call out. These modules won't have access to TypeORM.

Reply:
For the current module, it can just perform `db.X` functions itself by directly executing SQL commands. For other modules, it can use the following function to access their objects.
`call_mapper('{module_name}', '{entity_name}', 'cloud|db', 'create|read|update|delete', {}::json) -> json`